### ISSUE TITLE
IconButton widgets overlap when icons are of different sizes

### ISSUE THREAD

**Issue by flutternoob** [flutter/flutter#90272]
## Steps to Reproduce

I have tried the below code in DartPad. When the IconButton widgets have icons that are of different sizes, they overlap. However, they do not overlap when the icons are the same size. My code is give below along with the screenshots:

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(title: const Text("Overlapping IconButton Widgets")),
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Row(children: [
      IconButton(
        icon: Icon(
          Icons.replay_30_rounded,
          color: Colors.blue[900],
          size: 60,
        ),
        onPressed: () {},
      ),
      IconButton(
        icon: Icon(
          Icons.play_circle_fill_rounded,
          color: Colors.blue[900],
          size: 72,
        ),
        onPressed: () {},
      ),
      IconButton(
        icon: Icon(
          Icons.forward_30_rounded,
          color: Colors.blue[900],
          size: 60,
        ),
        onPressed: () {},
      ),
    ]);
  }
}
```
![Overlapping Icon Buttons - icons are of different sizes](https://user-images.githubusercontent.com/62204476/133804688-3efaaa8f-5a8d-4983-804b-722789c95f03.PNG)
![Overlapping Icon Buttons - icons are of same sizes](https://user-images.githubusercontent.com/62204476/133804729-b18b25fc-bc5a-4728-abec-e9b073d0d71d.PNG)

**Comment by darshankawar** [comment]
@flutternoob 
Can you try by adding `mainAxisAlignment: MainAxisAlignment.spaceEvenly` and see if it helps ?

<img width="327" alt="Screenshot 2021-09-20 at 12 14 45 PM" src="https://user-images.githubusercontent.com/67046386/133964818-2bc4c42a-eb1d-470f-863c-2810b0f5aaf1.png">

**Comment by darshankawar** [comment]
Thanks for the update. Verified this on latest stable and master and I see the same behavior.

<img width="162" alt="Screenshot 2021-09-21 at 6 32 35 PM" src="https://user-images.githubusercontent.com/67046386/134175858-cc03b2bc-1e6a-4536-bf37-c671c3a48848.png">


<details>
<summary> stable, master flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 2.5.1, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.5.1 at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (2 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.82


• No issues found!

[✓] Flutter (Channel master, 2.6.0-6.0.pre.160, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.6.0-6.0.pre.160 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a133306791 (13 hours ago), 2021-09-20 19:53:05 -0400
    • Engine revision 7e94275f73
    • Dart version 2.15.0 (build 2.15.0-126.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.82


• No issues found!




```

</details>

**Comment by HansMuller** [comment]
This problem can be avoided by overriding the default IconButton constraints and zeroing out the default padding:

```dart
Row(
  mainAxisSize: MainAxisSize.min,
  children: <Widget>[
    IconButton(
      padding: EdgeInsets.zero,
      constraints: BoxConstraints.tightFor(width: 60, height: 60),
      icon: Icon(
        Icons.replay_30_rounded,
        color: Colors.blue[900],
        size: 60,
      ),
      onPressed: () {},
    ),
    IconButton(
      padding: EdgeInsets.zero,
      constraints: BoxConstraints.tightFor(width: 72, height: 72),
      icon: Icon(
        Icons.play_circle_fill_rounded,
        color: Colors.blue[900],
        size: 72,
      ),
      onPressed: () {},
    ),
    IconButton(
      padding: EdgeInsets.zero,
      constraints: BoxConstraints.tightFor(width: 60, height: 60),
      icon: Icon(
        Icons.forward_30_rounded,
        color: Colors.blue[900],
        size: 60,
      ),
      onPressed: () {},
    ),
  ],
)
```

Still investigating why this doesn't happen by default.

**Comment by HansMuller** [comment]
I'm afraid that what we have here is  "working as intended".  The IconButton's `iconSize` parameter effectively overrides the icon's size. In this case, since iconSize is not being specified, what you're seeing is the default size (24) for all three icons. That's why they overlap. The [iconSize API doc](https://api.flutter.dev/flutter/material/IconButton/iconSize.html
) does cover this issue, although that's easy to miss (I did).

The fix is to simply remove the Icon size parameters and use the IconButton `iconSize` parameters instead.

I'm going to leave this issue open until we've documented this particular quirk a little more obviously.
